### PR TITLE
Telemetry end to end fixes

### PR DESCRIPTION
**Pull request by oldalton**
1. UI events not getting registered because of not using correct telemetry request ID
2. Silent request telemetry events not getting registered if forceRefresh == false
3. If user cancels UI immediately in interactive request, telemetry events don’t get logged
4. Show ui_cancelled = false in UI event, when user didn’t cancel interactive request
5. Update some value names to sync with other platforms (elapsed_time, token type etc)
6. UI behavior not set for interactive requests

### CODE REVIEW

File: MSAL/src/requests/MSALBaseRequest.m [hunk 1]
```diff
@@ -192,9 +188,11 @@ - (void)acquireToken:(nonnull MSALCompletionBlock)completionBlock
          {
              MSALErrorCode code = MSALErrorCodeForOAuthError(oauthError, MSALErrorInteractionRequired);
              
-             [self setEventError:event errorCode:error.code errorDomain:nil];
+             NSError *msalError = MSALCreateAndLogError(_parameters, code, oauthError, tokenResponse.subError, nil, __FUNCTION__, __LINE__, @"%@", tokenResponse.errorDescription);
```
Comment (oldalton) [inline MSAL/src/requests/MSALBaseRequest.m]: CREATE_LOG_ERROR ignores oauthError and subError. I can create an additional macro that would accept those as well, e.g CREATE_LOG_ERROR_WITH_SUBERRORS. Any better name suggestions?

**Comment by oldalton** [comment]
Ready for review again :)

File: MSAL/src/MSALError_Internal.m [hunk 2]
```diff
@@ -55,6 +55,9 @@
         STRING_CASE(MSALErrorNoViewController);
         STRING_CASE(MSALErrorInternal);
         STRING_CASE(MSALErrorUserNotFound);
+            
+        default:
+            return [NSString stringWithFormat:@"%ld", (long)code];
```
Comment (partnerinflight) [inline MSAL/src/MSALError_Internal.m]: nit: consider a string like "Unmapped Error %ld" instead

File: MSAL/src/cache/MSALTokenCacheAccessor.m [hunk 3]
```diff
@@ -106,7 +106,6 @@ - (BOOL)saveRefreshToken:(MSALRefreshTokenCacheItem *)rtItem
 
     BOOL result = [_dataSource addOrUpdateRefreshTokenItem:rtItem context:ctx error:error];
 
-    [event setStatus:result ? MSAL_TELEMETRY_VALUE_SUCCEEDED : MSAL_TELEMETRY_VALUE_FAILED];
```
Comment (partnerinflight) [inline MSAL/src/cache/MSALTokenCacheAccessor.m]: sorry, confused, why don't we want these?

Comment (oldalton) [inline MSAL/src/cache/MSALTokenCacheAccessor.m]: Ashish requested to remove status properties (copy-paste from telemetry channel: can we remove msal.status from the token events, its an artifact of OneDrive, i don't think it provides value).

File: MSAL/src/requests/MSALBaseRequest.m [hunk 4]
```diff
@@ -239,10 +237,8 @@ - (void)acquireToken:(nonnull MSALCompletionBlock)completionBlock
                                       user:atItem.user
                                     scopes:[tokenResponse.scope componentsSeparatedByString:@","]];
 
-         [event setClientId:_parameters.clientId];
```
Comment (partnerinflight) [inline MSAL/src/requests/MSALBaseRequest.m]: how come we're removing that?

File: MSAL/src/telemetry/MSALTelemetryAPIEvent.m [hunk 5]
```diff
@@ -125,10 +139,10 @@ - (void)setUIBehavior:(MSALUIBehavior)uiBehavior
 #pragma mark -
 #pragma mark log error
 
-- (void)setErrorCode:(NSInteger)errorCode
+- (void)setErrorCode:(MSALErrorCode)errorCode
 {
     self.errorInEvent = YES;
-    [self setProperty:MSAL_TELEMETRY_KEY_API_ERROR_CODE value:[NSString stringWithFormat:@"%ld", (long)errorCode]];
+    [self setProperty:MSAL_TELEMETRY_KEY_API_ERROR_CODE value:MSALStringForErrorCode(errorCode)];
```
Comment (RPangrle) [inline MSAL/src/telemetry/MSALTelemetryAPIEvent.m]: 👍

File: MSAL/test/unit/NSURL+MSALExtensionsTests.m [hunk 6]
```diff
@@ -68,4 +68,53 @@ - (void)testHostWithNoHost
     XCTAssertEqualObjects([urlWithNoHost msalHostWithPort], @"");
 }
 
+- (void)testScrubbedHttpPath_whenTenantlessAuthorityPath_shouldReturnSame
```
Comment (partnerinflight) [inline MSAL/test/unit/NSURL+MSALExtensionsTests.m]: love these!

**Pull request merged**